paddle/fluid/framework/operator.h

```diff
@@ -160,6 +160,10 @@ class OperatorBase {
                                  const platform::Place& place,
                                  const RuntimeContext& ctx) const {}
 
+  // Add the hooks
+  virtual void PreHook(const Scope& scope, const platform::Place& place);
```

Comment:
No need for hook. Just print the debug information at the right code place.

Reply:
remove them